packages/iTwinUI-react/src/core/Dialog/Dialog.tsx

```diff
@@ -37,20 +37,28 @@ export type DialogProps = {
  *   </Dialog.ButtonBar>
  * </Dialog>
  */
-export const Dialog = (props: DialogProps) => {
-  const { className, children, ...rest } = props;
+const DialogComponent = React.forwardRef<HTMLDivElement, DialogProps>(
+  (props, ref) => {
+    const { className, children, ...rest } = props;
 
-  useTheme();
+    useTheme();
 
-  return (
-    <div className={cx('iui-dialog', className)} role='dialog' {...rest}>
-      {children}
-    </div>
-  );
-};
+    return (
+      <div
+        className={cx('iui-dialog', className)}
+        role='dialog'
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </div>
+    );
+  },
+);
+export const Dialog = Object.assign(DialogComponent, {
```

Comment:
Creating a separate `const` will lose the jsdoc. You could either get rid of the intermediate `DialogComponent` (see `Carousel` for example) or move the jsdoc down to above `Dialog`.

Reply:
I've done that previously but didn't like that there is so many wrappers. But I also don't like moving jsdocs so just wrapped 😀